tests/test_modules.py

```diff
@@ -204,35 +205,36 @@ def test_install_profile_latest(self):
 
     def test_install_profile(self):
         self.test_enable_name_stream()
-        self.module_base.install(["httpd:2.4:1/default"])
+        self.module_base.install(["httpd:2.4:2/default"])
         self.base.resolve()
         expected = [
             "basesystem-11-3.noarch",
             "filesystem-3.2-40.x86_64",
             "glibc-2.25.90-2.x86_64",
             "glibc-common-2.25.90-2.x86_64",
-            "httpd-2.4.25-7.x86_64",
+            "httpd-2.4.25-8.x86_64",
             "libnghttp2-1.21.1-1.x86_64",  # expected behaviour, non-modular rpm pulled in
         ]
         self.assertInstalls(expected)
 
     def test_install_two_profiles(self):
         self.test_enable_name_stream()
 
-        self.module_base.install(["httpd:2.4:1/default", "httpd:2.4:1/doc"])
+        self.module_base.install(["httpd:2.4:2/default", "httpd:2.4:2/doc"])
         self.base.resolve()
         expected = [
             "basesystem-11-3.noarch",
             "filesystem-3.2-40.x86_64",
             "glibc-2.25.90-2.x86_64",
             "glibc-common-2.25.90-2.x86_64",
-            "httpd-2.4.25-7.x86_64",
-            "httpd-doc-2.4.25-7.x86_64",
+            "httpd-2.4.25-8.x86_64",
+            "httpd-doc-2.4.25-8.x86_64",
             "libnghttp2-1.21.1-1.x86_64",  # expected behaviour, non-modular rpm pulled in
         ]
         self.assertInstalls(expected)
 
     def test_install_two_profiles_different_versions(self):
+        """
```

Comment:
This test cannot work with the new implementation. In theory, it would be possible to add a workaround but the condition will be very complicated - think about algorithm that will join elements based on name, stream, and context (not version). Personally I don't feel comfortable to add another very complex logic to the code. I am glad, that we have this test, because it discovered additional change in the behavior with the new code.